grpc/proto/v1/generative.proto

```diff
@@ -0,0 +1,249 @@
+syntax = "proto3";
+
+package weaviate.v1;
+
+import "v1/base.proto";
+
+option go_package = "github.com/weaviate/weaviate/grpc/generated;protocol";
+option java_package = "io.weaviate.client.grpc.protocol.v1";
+option java_outer_classname = "WeaviateProtoGenerative";
+
+message GenerativeSearch {
+  message Single {
+    string prompt = 1;
+    bool debug = 2;
+    // only allow one at the beginning, but multiple in the future
+    repeated GenerativeProvider queries = 3;  
+  }
+  
+  message Grouped {
+    string task = 1;
+    optional TextArray properties = 2;
+  }
+
+  string single_response_prompt = 1 [deprecated = true];
+  string grouped_response_task = 2 [deprecated = true];
+  repeated string grouped_properties = 3 [deprecated = true];
+  Single single = 4;
+  Grouped grouped = 5;
+}
+  
+
+
+message GenerativeProvider {
+  bool return_metadata = 1;
+  oneof kind {
+    GenerativeAnthropic anthropic = 2;
+    GenerativeAnyscale anyscale = 3;
+    GenerativeAWS aws = 4;
+    GenerativeCohere cohere = 5;
+    GenerativeDummy dummy = 6;
+    GenerativeMistral mistral = 7;
+    GenerativeOctoAI octoai = 8;
+    GenerativeOllama ollama = 9;
+    GenerativeOpenAI openai = 10;
+    GenerativeGoogle google = 11;
+  }
```

Comment:
there's also a `generative-databricks` module, but we can add support for it in a separate PR